File: profesores.php
<?php
require 'menu.php';
require 'db.php';

// ── Inputs from tecnologia.php (or any subject page) ─────────────────────────
$materiaId  = (int)($_GET['materia'] ?? 0) ?: null;
$temas_raw  = trim($_GET['temas'] ?? '');
$temas      = $temas_raw
    ? array_map('trim', explode(',', $temas_raw))
    : ($_SESSION['temas_tecnologia'] ?? []);
$temas = array_slice($temas, 0, 5);

// ── Load materia info ─────────────────────────────────────────────────────────
$materia_info = $materiaId
    ? dbOne("SELECT materiaId, nombre, imagen FROM materias WHERE materiaId = :id", ['id' => $materiaId])
    : null;

// ── Query: teachers with active classes, soonest/most active first ────────────
// Sort: classes with active students first, then newest class
$whereMateria = $materiaId ? 'AND cp.materiaId = :mid' : '';
$params = $materiaId ? ['mid' => $materiaId] : [];

$clases = dbAll(
    "SELECT cp.claseId, cp.titulo, cp.descripcion, cp.precio_base, cp.codigo_moneda,
            cp.alumnos_min, cp.alumnos_max, cp.solo_yo, cp.created_at,
            u.usuarioId AS profId, u.nombre AS prof_nombre, u.avatar,
            u.calificacion, u.num_resenas, u.biografia,
            pa.nombre AS pais, pa.simbolo, pa.codigo_moneda AS mon_prof,
            m.nombre AS materia_nombre, m.imagen AS materia_img,
            COUNT(sc.sesionId) AS alumnos_activos
     FROM clases_programadas cp
     JOIN usuarios u          ON u.usuarioId  = cp.instructorId
     LEFT JOIN paises pa       ON pa.paisId    = u.pais_id
     LEFT JOIN materias m      ON m.materiaId  = cp.materiaId
     LEFT JOIN sesiones_clase sc ON sc.claseId = cp.claseId AND sc.fin IS NULL
     WHERE cp.activa = 1
     $whereMateria
     GROUP BY cp.claseId, cp.titulo, cp.descripcion, cp.precio_base, cp.codigo_moneda,
              cp.alumnos_min, cp.alumnos_max, cp.solo_yo, cp.created_at,
              u.usuarioId, u.nombre, u.avatar, u.calificacion, u.num_resenas, u.biografia,
              pa.nombre, pa.simbolo, pa.codigo_moneda, m.nombre, m.imagen
     ORDER BY alumnos_activos DESC, cp.created_at DESC",
    $params
);

// ── Convert slug to readable title ───────────────────────────────────────────
function slugToTitle(string $slug): string {
    return ucwords(str_replace('-', ' ', $slug));
}

// ── Map materiaId → subject page (breadcrumb / "change themes" links) ───────
$materia_page_map = [
    1  => 'matematicas.php', 2  => 'biologia.php',  3  => 'quimica.php',
    4  => 'fisica.php',      5  => 'historia.php',  6  => 'geografia.php',
    7  => 'literatura.php',  8  => 'idiomas.php',   9  => 'arte.php',
    10 => 'tecnologia.php',  11 => 'educacion_fisica.php',
];
$materiaPage = $materiaId ? ($materia_page_map[$materiaId] ?? 'materias.php') : 'materias.php';
?>

  <div class="container mt-10 pb-5">

    <!-- Breadcrumb -->
    <nav aria-label="breadcrumb" class="mb-3">
      <ol class="breadcrumb">
        <li class="breadcrumb-item"><a href="materias.php" class="text-secondary"><?= t('nav.materias') ?></a></li>
        <?php if ($materia_info): ?>
          <li class="breadcrumb-item">
            <a href="<?= $materiaPage ?>" class="text-secondary"><?= htmlspecialchars($materia_info['nombre']) ?></a>
          </li>
        <?php endif; ?>
        <li class="breadcrumb-item active text-white"><?= t('profesores.find_teacher') ?></li>
      </ol>
    </nav>

    <!-- Context banner ──────────────────────────────────────────────────────── -->
    <?php if ($materia_info || !empty($temas)): ?>
    <div class="card bg-dark border-secondary mb-4">
      <div class="card-body py-3 px-4">
        <div class="d-flex align-items-center gap-3 flex-wrap">
          <?php if ($materia_info && $materia_info['imagen']): ?>
            <img src="<?= htmlspecialchars($materia_info['imagen']) ?>"
                 style="width:48px;height:48px;object-fit:cover;border-radius:.4rem;" alt="">
          <?php endif; ?>
          <div>
            <?php if ($materia_info): ?>
              <div class="text-white fw-bold">
                <?= t('profesores.looking_for', ['subject' => htmlspecialchars($materia_info['nombre'])]) ?>
              </div>
            <?php endif; ?>
            <?php if (!empty($temas)): ?>
              <div class="text-secondary small mt-1">
                <span class="text-white"><?= t('profesores.selected_themes') ?></span>
                <?php foreach ($temas as $t): ?>
                  <span class="badge bg-secondary ms-1"><?= htmlspecialchars(slugToTitle($t)) ?></span>
                <?php endforeach; ?>
              </div>
            <?php endif; ?>
          </div>
          <a href="<?= $materia_info ? $materiaPage : 'materias.php' ?>"
             class="btn btn-sm btn-dark border-secondary ms-auto"><?= t('profesores.change_themes') ?></a>
        </div>
      </div>
    </div>
    <?php endif; ?>

    <div class="d-flex align-items-center justify-content-between mb-4">
      <div>
        <h2 class="text-white mb-0"><?= t('profesores.available_title') ?></h2>
        <p class="text-secondary mb-0 small">
          <?= t('profesores.sorted_by_activity') ?>
        </p>
      </div>
      <a href="buscar.php<?= $materiaId ? '?materia='.$materiaId : '' ?>"
         class="btn btn-dark border-secondary"><?= t('profesores.all_classes') ?></a>
    </div>

    <!-- ── Teacher / Class cards ─────────────────────────────────────────────── -->
    <?php if (empty($clases)): ?>
      <div class="alert alert-dark border border-secondary text-center py-4">
        <p class="text-secondary mb-2"><?= t('profesores.no_live') ?></p>
        <p class="text-secondary small mb-3"><?= t('profesores.be_first') ?></p>
        <a href="crear_clase.php<?= $materiaId ? '?materia='.$materiaId : '' ?>"
           class="btn btn-secondary"><?= t('profesores.post_class') ?></a>
      </div>

      <!-- ── Demo / seed teachers (static fallback so the page is never empty) ── -->
      <p class="text-secondary small mb-3 mt-4"><?= t('profesores.registered_teachers') ?></p>
      <div class="row g-3">
        <?php
        $demo = [
          ['nombre'=>'Alexander V.', 'rol'=>'Director',   'rating'=>4.8, 'reseñas'=>123, 'time'=>'00:05', 'pais'=>'Chile'],
          ['nombre'=>'Liam S.',      'rol'=>'Instructor', 'rating'=>4.2, 'reseñas'=>98,  'time'=>'10:00', 'pais'=>'Mexico'],
          ['nombre'=>'Elena R.',     'rol'=>'Assistant',  'rating'=>3.5, 'reseñas'=>47,  'time'=>'15:00', 'pais'=>'Argentina'],
          ['nombre'=>'Marcus T.',    'rol'=>'Researcher', 'rating'=>4.9, 'reseñas'=>210, 'time'=>'05:00', 'pais'=>'Colombia'],
          ['nombre'=>'Sophia L.',    'rol'=>'Adviser',    'rating'=>3.8, 'reseñas'=>61,  'time'=>'08:00', 'pais'=>'Peru'],
          ['nombre'=>'Diana P.',     'rol'=>'Director',   'rating'=>4.5, 'reseñas'=>175, 'time'=>'12:00', 'pais'=>'Brazil'],
        ];
        foreach ($demo as $d):
          $stars = str_repeat('★', round($d['rating'])) . str_repeat('☆', 5-round($d['rating']));
        ?>
        <div class="col-sm-6 col-lg-4">
          <div class="card bg-dark border-secondary h-100">
            <div class="card-body">
              <div class="d-flex align-items-center gap-3 mb-3">
                <div class="rounded-circle bg-secondary d-flex align-items-center justify-content-center flex-shrink-0"
                     style="width:52px;height:52px;font-size:1.4rem;">👤</div>
                <div>
                  <div class="text-white fw-bold"><?= $d['nombre'] ?></div>
                  <div class="text-secondary small"><?= t('profesores.role_' . strtolower($d['rol'])) ?> · 📍 <?= $d['pais'] ?></div>
                </div>
              </div>
              <div class="text-warning mb-2" style="font-size:.85rem;">
                <?= $stars ?> <span class="text-secondary">(<?= $d['reseñas'] ?> <?= t('profesores.reviews') ?>)</span>
              </div>
              <div class="d-flex align-items-center gap-2 text-secondary small">
                <span class="badge bg-warning text-dark"><?= t('profesores.starts_in', ['time' => $d['time']]) ?></span>
                <?php if ($materia_info): ?>
                  <span class="badge bg-secondary"><?= htmlspecialchars($materia_info['nombre']) ?></span>
                <?php endif; ?>
              </div>
            </div>
            <div class="card-footer bg-dark border-secondary">
              <a href="crear_clase.php" class="btn btn-dark border-secondary w-100 text-secondary" disabled>
                <?= t('profesores.no_live_class') ?>
              </a>
            </div>
          </div>
        </div>
        <?php endforeach; ?>
      </div>

    <?php else: ?>

      <div class="row g-3" id="teachers-grid">
        <?php foreach ($clases as $c):
          $spots   = $c['alumnos_max'] - $c['alumnos_activos'];
          $full    = $spots <= 0;
          $active  = $c['alumnos_activos'] > 0;
          $stars   = str_repeat('★', min(5, (int)round($c['calificacion'])))
                   . str_repeat('☆', max(0, 5 - (int)round($c['calificacion'])));
        ?>
        <div class="col-sm-6 col-lg-4">
          <div class="card bg-dark border-secondary h-100 <?= $active ? 'border-primary' : '' ?>"
               style="<?= $active ? 'border-color:var(--p)!important;' : '' ?>">

            <?php if ($active): ?>
              <div class="card-header border-0 py-2 px-3"
                   style="background:color-mix(in srgb, var(--p) 15%, transparent);">
                <div class="d-flex justify-content-between align-items-center">
                  <span class="text-primary small fw-bold"><?= t('profesores.live_now', ['count' => $c['alumnos_activos']]) ?></span>
                  <div class="text-center">
                    <div class="timer-label text-white small"><?= t('profesores.live_class_label') ?></div>
                    <div class="timer text-primary">00:00</div>
                  </div>
                </div>
              </div>
            <?php else: ?>
              <div class="card-header border-0 py-2 px-3 bg-dark">
                <span class="text-secondary small"><?= t('profesores.posted', ['time' => timeAgo($c['created_at'])]) ?></span>
              </div>
            <?php endif; ?>

            <div class="card-body">
              <!-- Subject badge -->
              <?php if ($c['materia_nombre']): ?>
                <span class="badge bg-secondary mb-2"><?= htmlspecialchars($c['materia_nombre']) ?></span>
              <?php endif; ?>

              <h5 class="card-title text-white mb-1"><?= htmlspecialchars($c['titulo'] ?: t('profesores.open_class')) ?></h5>
              <?php if ($c['descripcion']): ?>
                <p class="text-secondary small mb-3"><?= htmlspecialchars(mb_substr($c['descripcion'], 0, 100)) ?>…</p>
              <?php endif; ?>

              <!-- Teacher row -->
              <div class="d-flex align-items-center gap-2 mb-3">
                <?php if ($c['avatar']): ?>
                  <img src="<?= htmlspecialchars($c['avatar']) ?>"
                       class="rounded-circle flex-shrink-0"
                       width="44" height="44" style="object-fit:cover;" alt="">
                <?php else: ?>
                  <div class="rounded-circle bg-secondary d-flex align-items-center justify-content-center flex-shrink-0"
                       style="width:44px;height:44px;font-size:1.2rem;">👤</div>
                <?php endif; ?>
                <div class="min-w-0">
                  <div class="text-white fw-semibold text-truncate"><?= htmlspecialchars($c['prof_nombre']) ?></div>
                  <div class="text-warning" style="font-size:.75rem;">
                    <?= $stars ?>
                    <span class="text-secondary">(<?= (int)$c['num_resenas'] ?>)</span>
                  </div>
                  <?php if ($c['pais']): ?>
                    <div class="text-secondary" style="font-size:.72rem;">📍 <?= htmlspecialchars($c['pais']) ?></div>
                  <?php endif; ?>
                </div>
              </div>

              <!-- Price + spots -->
              <div class="d-flex justify-content-between align-items-end">
                <div>
                  <span class="text-white fw-bold fs-5">
                    <?= htmlspecialchars($c['simbolo'] ?? '$') ?><?= number_format($c['precio_base'], 2) ?>
                  </span>
                  <span class="text-secondary small ms-1"><?= htmlspecialchars($c['codigo_moneda']) ?><?= t('profesores.per_session') ?></span>
                  <div class="text-secondary" style="font-size:.72rem;"><?= t('profesores.billed_local') ?></div>
                </div>
                <span class="badge <?= $full ? 'bg-danger' : ($active ? 'bg-success' : 'bg-secondary') ?>">
                  <?= $full ? t('profesores.full') : ($active ? ($spots === 1 ? t('profesores.spot_left', ['n' => $spots]) : t('profesores.spots_left', ['n' => $spots])) : t('profesores.open')) ?>
                </span>
              </div>
            </div>

            <div class="card-footer bg-dark border-secondary">
              <?php if ($full): ?>
                <button class="btn btn-secondary w-100" disabled><?= t('profesores.class_full') ?></button>
              <?php else: ?>
                <a href="pre_sala.php?clase=<?= $c['claseId'] ?>&from=explorar"
                   class="btn <?= $active ? 'btn-success' : 'btn-dark border-secondary' ?> w-100 fw-semibold">
                  <?= $active ? t('profesores.join_live') : t('profesores.start_session') ?>
                </a>
              <?php endif; ?>
            </div>

          </div>
        </div>
        <?php endforeach; ?>
      </div>

    <?php endif; ?>

  </div>

  <footer class="mastfoot mt-5">
    <div class="inner float-end">
      <p><?= t('general.footer_text', ['bootstrap' => '<a href="https://getbootstrap.com/">Bootstrap</a>', 'author' => '<a href="https://www.facebook.com/rodrigo.alejandro.1848816?locale=es_LA">@RodrigoConejeros</a>']) ?></p>
    </div>
  </footer>

  <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/js/bootstrap.bundle.min.js" integrity="sha384-YvpcrYf0tY3lHB60NNkmXc5s9fDVZLESaAA55NDzOxhy9GkcIdslK1eN7N6jIeHz" crossorigin="anonymous"></script>
  <script src="https://code.jquery.com/jquery-3.7.1.min.js" integrity="sha256-/JqT3SQfawRcv/BIHPThkBvs0OEvtFFmqPF/lYI/Cxo=" crossorigin="anonymous"></script>

</body>
</html>

File: go/internal/web/profesores.go
package web

import (
	"fmt"
	"net/http"
	"strings"

	"classexpress/internal/i18n"
	"classexpress/internal/store"
)

// materiaPageMap maps materiaId -> subject landing page (breadcrumb links).
var materiaPageMap = map[int64]string{
	1:  "matematicas.php",
	2:  "biologia.php",
	3:  "quimica.php",
	4:  "fisica.php",
	5:  "historia.php",
	6:  "geografia.php",
	7:  "literatura.php",
	8:  "idiomas.php",
	9:  "arte.php",
	10: "tecnologia.php",
	11: "educacion_fisica.php",
}

// slugToTitle mirrors slugToTitle(): ucwords(str_replace('-', ' ', slug)).
func slugToTitle(slug string) string {
	parts := strings.Fields(strings.ReplaceAll(slug, "-", " "))
	for i, w := range parts {
		if w == "" {
			continue
		}
		parts[i] = strings.ToUpper(w[:1]) + w[1:]
	}
	return strings.Join(parts, " ")
}

// spotsText renders the class availability badge text (parity with profesores.php).
func spotsText(lang string, spots int64, active bool) string {
	switch {
	case spots <= 0:
		return i18n.T(lang, "profesores.full", nil)
	case active && spots == 1:
		return i18n.T(lang, "profesores.spot_left", map[string]string{"n": "1"})
	case active:
		return i18n.T(lang, "profesores.spots_left", map[string]string{"n": fmt.Sprint(spots)})
	default:
		return i18n.T(lang, "profesores.open", nil)
	}
}

type profClase struct {
	ClaseId       int64
	ProfesorID    int64
	Titulo        string
	Descripcion   string
	MateriaNombre string
	ProfNombre    string
	Avatar        string
	HasAvatar     bool
	Pais          string
	Calificacion  float64
	NumResenas    int64
	Simbolo       string
	CodigoMoneda  string
	Precio        string
	Activos       int64
	Spots         int64
	Full          bool
	Active        bool
	Stars         string
	Posted        string
	LiveNow       string
	SpotsText     string
}

type profDemo struct {
	Nombre       string
	Rol          string
	Rating       float64
	Resenas      int64
	Time         string
	Pais         string
	Stars        string
	StartsIn     string
	MateriaNombre string
}

// stars renders the ★☆ string from a rating (str_repeat pattern in PHP).
func stars(rating float64) string {
	n := int(rating + 0.5)
	if n < 0 {
		n = 0
	}
	if n > 5 {
		n = 5
	}
	return strings.Repeat("\u2605", n) + strings.Repeat("\u2606", 5-n)
}

// HandleProfesores ports profesores.php (subject teacher/class finder).
func (p *Pages) HandleProfesores(w http.ResponseWriter, r *http.Request) {
	ctx := r.Context()
	s := SessionFrom(ctx)
	if s == nil {
		serverError(w, errNoSession)
		return
	}
	if !p.GuardPage(w, r, s) {
		return
	}
	lang := p.ResolveLang(s, r)

	page := CurrentPage(r)
	nav, stop := p.MenuData(w, r, s, page, lang)
	if stop {
		return
	}

	materiaId := store.Int(r.URL.Query().Get("materia"))
	var temas []string
	if raw := strings.TrimSpace(r.URL.Query().Get("temas")); raw != "" {
		for _, t := range strings.Split(raw, ",") {
			if t = strings.TrimSpace(t); t != "" {
				temas = append(temas, t)
			}
		}
	} else if sess := s.Get("temas_tecnologia"); sess != "" {
		for _, t := range strings.Split(sess, ",") {
			if t = strings.TrimSpace(t); t != "" {
				temas = append(temas, t)
			}
		}
	}
	if len(temas) > 5 {
		temas = temas[:5]
	}
	temasTitles := make([]string, 0, len(temas))
	for _, t := range temas {
		temasTitles = append(temasTitles, slugToTitle(t))
	}

	var materiaInfo map[string]any
	if materiaId > 0 {
		var err error
		materiaInfo, err = p.DB.QueryOne(ctx,
			"SELECT materiaId, nombre, imagen FROM materias WHERE materiaId = ?", materiaId)
		if err != nil {
			materiaInfo = nil
		}
	}
	materiaNombre := ""
	if materiaInfo != nil {
		materiaNombre = store.Str(materiaInfo["nombre"])
	}

	var (
		clases  []profClase
		demo    []profDemo
		hasRows bool
	)
	rows, err := p.DB.QueryAll(ctx,
		`SELECT cp.claseId, cp.titulo, cp.descripcion, cp.precio_base, cp.codigo_moneda,
		        cp.alumnos_min, cp.alumnos_max, cp.solo_yo, cp.created_at,
		        u.usuarioId AS profId, u.nombre AS prof_nombre, u.avatar,
		        u.calificacion, u.num_resenas, u.biografia,
		        pa.nombre AS pais, pa.simbolo, pa.codigo_moneda AS mon_prof,
		        m.nombre AS materia_nombre, m.imagen AS materia_img,
		        COUNT(sc.sesionId) AS alumnos_activos
		 FROM clases_programadas cp
		 JOIN usuarios u          ON u.usuarioId  = cp.instructorId
		 LEFT JOIN paises pa       ON pa.paisId    = u.pais_id
		 LEFT JOIN materias m      ON m.materiaId  = cp.materiaId
		 LEFT JOIN sesiones_clase sc ON sc.claseId = cp.claseId AND sc.fin IS NULL
		 WHERE cp.activa = 1`+claseMateriaWhere(materiaId)+`
		 GROUP BY cp.claseId, cp.titulo, cp.descripcion, cp.precio_base, cp.codigo_moneda,
		          cp.alumnos_min, cp.alumnos_max, cp.solo_yo, cp.created_at,
		          u.usuarioId, u.nombre, u.avatar, u.calificacion, u.num_resenas, u.biografia,
		          pa.nombre, pa.simbolo, pa.codigo_moneda, m.nombre, m.imagen
		 ORDER BY alumnos_activos DESC, cp.created_at DESC`,
		claseMateriaArgs(materiaId)...)
	if err == nil {
		hasRows = len(rows) > 0
		for _, row := range rows {
			claseId := store.Int(row["claseId"])
			activos := store.Int(row["alumnos_activos"])
			max := store.Int(row["alumnos_max"])
			spots := max - activos
			active := activos > 0
			rating := store.Float(row["calificacion"])
		clases = append(clases, profClase{
			ClaseId:       claseId,
			ProfesorID:    store.Int(row["profId"]),
			Titulo:        store.Str(row["titulo"]),
			Descripcion:   truncUTF8(store.Str(row["descripcion"]), 100),
			MateriaNombre: store.Str(row["materia_nombre"]),
			ProfNombre:    store.Str(row["prof_nombre"]),
			Avatar:        store.Str(row["avatar"]),
			HasAvatar:     store.Str(row["avatar"]) != "",
			Pais:          store.Str(row["pais"]),
			Calificacion:  rating,
			NumResenas:    store.Int(row["num_resenas"]),
			Simbolo:       store.Str(row["simbolo"]),
			CodigoMoneda:  store.Str(row["codigo_moneda"]),
			Precio:        fmt.Sprintf("%.2f", store.Float(row["precio_base"])),
			Activos:       activos,
			Spots:         spots,
			Full:          spots <= 0,
			Active:        active,
			Stars:         stars(rating),
			Posted:        TimeAgo(row["created_at"]),
			LiveNow:       i18n.T(lang, "profesores.live_now", map[string]string{"count": fmt.Sprint(activos)}),
			SpotsText:     spotsText(lang, spots, active),
		})
	}
	}

	if !hasRows {
		// Demo / seed teachers (static fallback so the page is never empty).
		seed := []struct {
			nombre, rol, pais, time string
			rating                  float64
			resenas                 int64
		}{
			{"Alexander V.", "Director", "Chile", "00:05", 4.8, 123},
			{"Liam S.", "Instructor", "Mexico", "10:00", 4.2, 98},
			{"Elena R.", "Assistant", "Argentina", "15:00", 3.5, 47},
			{"Marcus T.", "Researcher", "Colombia", "05:00", 4.9, 210},
			{"Sophia L.", "Adviser", "Peru", "08:00", 3.8, 61},
			{"Diana P.", "Director", "Brazil", "12:00", 4.5, 175},
		}
		for _, d := range seed {
			demo = append(demo, profDemo{
				Nombre:        d.nombre,
				Rol:           i18n.T(lang, "profesores.role_"+strings.ToLower(d.rol), nil),
				Rating:        d.rating,
				Resenas:       d.resenas,
				Time:          d.time,
				Pais:          d.pais,
				Stars:         stars(d.rating),
				StartsIn:      i18n.T(lang, "profesores.starts_in", map[string]string{"time": d.time}),
				MateriaNombre: materiaNombre,
			})
		}
	}

	materiaPage := "materias.php"
	if materiaId > 0 {
		if m, ok := materiaPageMap[materiaId]; ok {
			materiaPage = m
		}
	}

	data := map[string]any{
		"Lang":          lang,
		"NavData":       nav,
		"MateriaId":     materiaId,
		"MateriaInfo":   materiaInfo != nil,
		"MateriaNombre": materiaNombre,
		"MateriaImagen": "",
		"MateriaPage":   materiaPage,
		"Temas":         temas,
		"TemasTitles":   temasTitles,
		"HasContext":    materiaInfo != nil || len(temas) > 0,
		"LookingFor":    i18n.T(lang, "profesores.looking_for", map[string]string{"subject": materiaNombre}),
		"BuscarHref":    "buscar.php" + mapIf(materiaId > 0, "?materia="+fmt.Sprint(materiaId), ""),
		"CrearHref":     "crear_clase.php" + mapIf(materiaId > 0, "?materia="+fmt.Sprint(materiaId), ""),
		"HasRows":       hasRows,
		"Clases":        clases,
		"Demo":          demo,
	}
	if materiaInfo != nil {
		data["MateriaImagen"] = store.Str(materiaInfo["imagen"])
	}
	if err := p.Templates.Render(w, "profesores", p, s, lang, data); err != nil {
		serverError(w, err)
	}
}

func claseMateriaWhere(id int64) string {
	if id > 0 {
		return " AND cp.materiaId = ?"
	}
	return ""
}

func claseMateriaArgs(id int64) []any {
	if id > 0 {
		return []any{id}
	}
	return nil
}

func mapIf(cond bool, a, b string) string {
	if cond {
		return a
	}
	return b
}

// truncUTF8 truncates a UTF-8 string to n runes (mb_substr fallback).
func truncUTF8(s string, n int) string {
	r := []rune(s)
	if len(r) <= n {
		return s
	}
	return string(r[:n])
}
